bot/resources/tags/loop-remove.md

```diff
@@ -0,0 +1,31 @@
+---
+embed:
+    title: "Removing items inside a for loop"
+---
+Avoid removing items from a collection, such as a list, as you iterate that collection in a `for` loop:
+```py
+data = [1, 2, 3, 4]
+for item in data:
+    data.remove(item)
+print(data)  # [2, 4] <-- every OTHER item was removed!
+```
+`for` loops track the index of the current item with a kind of pointer. Removing an element causes all other elements to shift, but the pointer is not changed:
```

Comment:
Technically it's not the for loop tracking it, it's the list's iterator. I don't think this really matters, but I also think it's not too hard to word it to avoid suggesting that. I'm also not sure how much "pointer" will mean to someone without knowledge in a language like C. Maybe something along the lines of this would be clearer?
```suggestion
While iterating with a for loop, an index tracks the current position in the list. If the list is modified, this index may no longer refer to the same element, causing elements to be repeated or skipped.
```